src/todos/todos.controller.ts

```diff
@@ -7,31 +7,25 @@ import {
   Param,
   Patch,
   Post,
+  UsePipes,
+  ValidationPipe,
 } from '@nestjs/common'
-import { Types } from 'mongoose'
 
 import { TODOS_ROUTE } from 'src/common/constants/routing.constants'
-import { ValidationError } from 'src/common/errors/errors'
 import { TodosService } from './todos.service'
+import { CreateTodoDto } from './dto/create-todo.dto'
+import { UpdateTodoDto } from './dto/update-todo.dto'
+import { IdParamDto } from './dto/id-param.dto'
 import type { Todo } from './schemas/todos.schema'
 
 @Controller(TODOS_ROUTE)
 export class TodosController {
   constructor(private readonly todosService: TodosService) {}
 
   @Post()
-  async create(@Body() todo: any): Promise<Todo> {
-    if (
-      typeof todo !== 'object' ||
-      typeof todo.title !== 'string' ||
-      !todo.title.trim()
-    ) {
-      throw new ValidationError(
-        'Invalid todo. Title is required and should be a non-empty string.',
-      )
-    }
-
-    return await this.todosService.create(todo)
+  @UsePipes(ValidationPipe)
```

Comment:
Is it possible to apply pipe to the whole controller instead of applying it to each route separately?

Reply:
Sure, fixed